### PR TITLE
[2.7] Update Provisioning Input & Release Upgrade fields'  tags and Support OS & Upgrade HA Jenkinsfiles

### PR DESCRIPTION
**Pull request by caliskanugur**
## Problem

- Because of multiple unmarshal and marshal while updating the generating configuration, glue codes are generating zero value fields, causing some nil checks in the tests to pass. 
- Support matrix OS can't point to different branches and repos.
- Release upgrade HA's last step is dependent on generated patched configs rather than initial configurations, this was causing not to provision the new clusters on the upgraded version if the initial cluster provision failed.
 
## Solution

- Added omit empty tags to provisioning input fields
- Updated support OS matrix and release HA upgrade Jenkinsfiles' repo logics to unify with e2e Jenkinsfile.
- Updated release upgrade HA's last loop to use the initially generated configuration files. 
- Updated release upgrade HA configuration matrix fields to have JSON tags

 
### Automated Testing

* Test types added/modified:
    * Validation (Go Framework)

### CODE REVIEW

**Comment by caliskanugur** [comment]
Todos:

- [ ] Rebase and cherry-pick the forwardport again after provisioning input changes are merged to 2.7.

File: tests/v2/validation/Jenkinsfile_support_matrix_os [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+      job_names = job_name.split('/')
+      job_name = job_names[job_names.size() - 1] 
+    }
+    def testContainer = "${job_name}${env.BUILD_NUMBER}_test"
+    def imageName = "rancher-validation-${job_name}${env.BUILD_NUMBER}"
+    def testsDir = "github.com/rancher/rancher/tests/v2/validation/${env.TEST_PACKAGE}"
+    def testResultsOut = "results.xml"
+    def envFile = ".env"
+    def rancherConfig = "rancher_env.config"
+    def branch = "release/v2.7"
```
Comment (izaac) [inline tests/v2/validation/Jenkinsfile_support_matrix_os:15]: This `branch` has to match the current and a separate file for v2.8?

File: tests/v2/validation/Jenkinsfile_support_matrix_os [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    if ("${env.BRANCH}" != "null" && "${env.BRANCH}" != "") {
+      branch = "${env.BRANCH}"
+    }
+    def rancherRepo = scm.getUserRemoteConfigs()[0].getUrl()
```
Comment (izaac) [inline tests/v2/validation/Jenkinsfile_support_matrix_os:19]: We most likely never run into this issue as there's always going to be a value in `getUserRemoteConfigs()[0]`. But do you think a check for `null` will make this safer or is it just redundant?

**Review by izaac — COMMENTED** [review]
Looks good just a couple notes.

File: tests/v2/validation/Jenkinsfile_support_matrix_os (hunk 1 above)
Comment (caliskanugur) [reply tests/v2/validation/Jenkinsfile_support_matrix_os:15]: So this PR is pointing to the 2.7 branch, should it be the 2.8? Not a separate file but more like its backward port just to unify branches. This file doesn't exist on 2.7, and we don't have a backport for it, so I added this file to the PR.

File: tests/v2/validation/Jenkinsfile_support_matrix_os (hunk 2 above)
Comment (caliskanugur) [reply tests/v2/validation/Jenkinsfile_support_matrix_os:19]: I don't have a strong opinion, but for that, the user would need to delete the default repo from the repo input, correct me if I am wrong. Also, this is unifying with Jenkinsfile.e2e. I can add this check to this, and upgrade HA and e2e Jenkinsfiles if agreed

File: tests/v2/validation/Jenkinsfile_support_matrix_os (hunk 1 above)
Comment (caliskanugur) [reply tests/v2/validation/Jenkinsfile_support_matrix_os:15]: So to catch this, this PR will be stale until this: https://github.com/rancher/rancher/pull/41061 PR is merged. That way we can have the same history for both branches. After a rebase, this PR will only point to the git checkout part like the 2.8 branch.

cc: @igomez06

**Review by izaac — APPROVED** [review]
LGTM

**Review by igomez06 — APPROVED** [review]
LGTM

**Review by markusewalker — APPROVED** [review]
LGTM

**Pull request merged**